core/src/main/java/org/infinispan/functional/impl/MetaParams.java

```diff
@@ -288,6 +290,25 @@ void merge(MetaParams other) {
       return new It();
    }
 
+   @Override
+   public Spliterator<MetaParam<?>> spliterator() {
+      return Spliterators.spliterator(iterator(), length, Spliterator.DISTINCT | Spliterator.NONNULL);
+   }
+
+   public MetaParam.Writable[] toWritableMetas() {
+      int writable = 0;
```

Comment:
Yes, but none should block each other. The current solution is fine, if we decide to not include the setter.